src/index.ts

```diff
@@ -13,55 +13,44 @@ let counter = 0
 //  - Generates short results
 //
 // This is not a serialization function, and the result is not guaranteed to be
-// parsible.
+// parsable.
 export default function stableHash(arg: any): string {
   const type = typeof arg
   const constructor = arg && arg.constructor
   const isDate = constructor == Date
 
-  let result: any
-  let index: any
-
   if (Object(arg) === arg && !isDate && constructor != RegExp) {
     // Object/function, not null/date/regexp. Use WeakMap to store the id first.
     // If it's already hashed, directly return the result.
-    result = table.get(arg)
+    let result = table.get(arg)
     if (result) return result
-
-    // Store the hash first for circular reference detection before entering the
-    // recursive `stableHash` calls.
-    // For other objects like set and map, we use this id directly as the hash.
-    result = ++counter + "~"
-    table.set(arg, result)
```

Comment:
This is necessary, otherwise recursive objects won't work:

```
const a = []
a[0] = a

stableHash(a) // stack overflow
```

Reply:
@shuding Hey! I guess it's hard to make a review for these changes, but I didn't change original logic of code
<img width="1208" alt="image" src="https://user-images.githubusercontent.com/7361780/183475096-6458471e-7951-4c9a-bd60-1943a6fdd9b2.png">

I only replaced highlighted lines by single one before `return result`